From 986230988524ada44d2a49cadb37c6ee3edd8bce Mon Sep 17 00:00:00 2001
From: caiyinyu <caiyinyu@loongson.cn>
Date: Wed, 12 Jan 2022 17:38:48 +0800
Subject: [PATCH 31/32] add math-use-builtins-sqrt.h and modify related code.
 p13

	deleted:    sysdeps/loongarch/fpu/e_sqrt.c
	deleted:    sysdeps/loongarch/fpu/e_sqrtf.c
	new file:   sysdeps/loongarch/fpu/math-use-builtins-sqrt.h
	modified:   ../sysdeps/unix/sysv/linux/loongarch/lp64/libm.abilist
---
 sysdeps/loongarch/fpu/e_sqrt.c                | 26 -------------------
 sysdeps/loongarch/fpu/e_sqrtf.c               | 26 -------------------
 .../loongarch/fpu/math-use-builtins-sqrt.h    |  4 +++
 .../sysv/linux/loongarch/lp64/libm.abilist    |  2 --
 4 files changed, 4 insertions(+), 54 deletions(-)
 delete mode 100644 sysdeps/loongarch/fpu/e_sqrt.c
 delete mode 100644 sysdeps/loongarch/fpu/e_sqrtf.c
 create mode 100644 sysdeps/loongarch/fpu/math-use-builtins-sqrt.h

diff --git a/sysdeps/loongarch/fpu/e_sqrt.c b/sysdeps/loongarch/fpu/e_sqrt.c
deleted file mode 100644
index 352e1cd30f..0000000000
--- a/sysdeps/loongarch/fpu/e_sqrt.c
+++ /dev/null
@@ -1,26 +0,0 @@
-/* Double-precision floating point square root.
-   Copyright (C) 2021 Free Software Foundation, Inc.
-   This file is part of the GNU C Library.
-
-   The GNU C Library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2.1 of the License, or (at your option) any later version.
-
-   The GNU C Library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with the GNU C Library.  If not, see
-   <https://www.gnu.org/licenses/>.  */
-
-double
-__ieee754_sqrt (double x)
-{
-  double z;
-  __asm__("fsqrt.d %0,%1" : "=f"(z) : "f"(x));
-  return z;
-}
-strong_alias (__ieee754_sqrt, __sqrt_finite)
diff --git a/sysdeps/loongarch/fpu/e_sqrtf.c b/sysdeps/loongarch/fpu/e_sqrtf.c
deleted file mode 100644
index 950156e9f8..0000000000
--- a/sysdeps/loongarch/fpu/e_sqrtf.c
+++ /dev/null
@@ -1,26 +0,0 @@
-/* Single-precision floating point square root.
-   Copyright (C) 2021 Free Software Foundation, Inc.
-   This file is part of the GNU C Library.
-
-   The GNU C Library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2.1 of the License, or (at your option) any later version.
-
-   The GNU C Library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with the GNU C Library.  If not, see
-   <https://www.gnu.org/licenses/>.  */
-
-float
-__ieee754_sqrtf (float x)
-{
-  float z;
-  __asm__("fsqrt.s %0,%1" : "=f"(z) : "f"(x));
-  return z;
-}
-strong_alias (__ieee754_sqrtf, __sqrtf_finite)
diff --git a/sysdeps/loongarch/fpu/math-use-builtins-sqrt.h b/sysdeps/loongarch/fpu/math-use-builtins-sqrt.h
new file mode 100644
index 0000000000..e94c915ba6
--- /dev/null
+++ b/sysdeps/loongarch/fpu/math-use-builtins-sqrt.h
@@ -0,0 +1,4 @@
+#define USE_SQRT_BUILTIN 1
+#define USE_SQRTF_BUILTIN 1
+#define USE_SQRTL_BUILTIN 0
+#define USE_SQRTF128_BUILTIN 0
diff --git a/sysdeps/unix/sysv/linux/loongarch/lp64/libm.abilist b/sysdeps/unix/sysv/linux/loongarch/lp64/libm.abilist
index 63ab4b261c..6f0e1c2100 100644
--- a/sysdeps/unix/sysv/linux/loongarch/lp64/libm.abilist
+++ b/sysdeps/unix/sysv/linux/loongarch/lp64/libm.abilist
@@ -17,8 +17,6 @@ GLIBC_2.35 __signbit F
 GLIBC_2.35 __signbitf F
 GLIBC_2.35 __signbitl F
 GLIBC_2.35 __signgam D 0x4
-GLIBC_2.35 __sqrt_finite F
-GLIBC_2.35 __sqrtf_finite F
 GLIBC_2.35 __sqrtl_finite F
 GLIBC_2.35 acos F
 GLIBC_2.35 acosf F
-- 
2.35.1

